File: 0011-container-with-most-water/0011-container-with-most-water.c
#define Max(a , b)  (a > b ? a : b);
#define Min(a , b)  (a < b ? a : b);

int maxArea(int* height, int heightSize) {
    int left = 0;
    int right = heightSize - 1;
    int current_water = 0;
    int result = 0;

    while(left <= right){
        current_water = (right - left) * Min(height[right] , height[left]);
        result = Max(result , current_water);

        if(height[left] < height[right]){
            left++;
        }else{
            right--;
        }
    }

    return result;
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        ans = 0
        while left < right:
            curr_width = right - left
            curr_height = min(height[left] , height[right])
            temp = curr_width * curr_height

            if temp > ans:
                ans = temp

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return ans